src/scripts/marcRecordUi.js

```diff
@@ -336,55 +337,265 @@ export function filterField(field) {
+  if (!isDataFieldTag(value)) {
+    return value.length;
+  }
+  /*
+  if (isDataFieldTag(value)) {
+    // Data field: tag(3) + indicators (2) + prefix.length ('$$'==2). Don't protect subfield code (1)
+    return 5 + articleEditorSettings.subfieldCodePrefix.length;
+  }
+    */
+  // Control field: tag (3) + empty (2)
+  return 5;
+}
+
+function getOvertypeLength(event, inputText, fieldAsString, position) { // position means position when text has been added or removed
+  if (!inputText) {
+    if (event.inputType === 'deleteContentBackward') {
+      return -1;
+    }
+
+    return 0;
+  }
+
+  if (position === fieldAsString.length) {
+    // If there's nothing on the right, there's no need for overtype :-)
+    return 0;
+  }
+  const jumpSize = inputText.length;
+  const startPosition = position - jumpSize;
+  //const tailLength = fieldAsString.length.position;
+  //const originalLength = startPosition + tailLength;
+  let i=0;
+  const protectedAreaSize = getProtectedAreaSize(fieldAsString);
+  while (i < jumpSize && startPosition+i < protectedAreaSize) {
+    i++;
+  }
+  // console.log(`START: ${startPosition}, LEN: ${jumpSize}, OVERTYPE: ${i} char(s)`);
+  return i;
+}
+
+function editorHandleInput(event, settings) {
+  var elem = event.currentTarget;
+  console.log(`INPUT EVENT ${event.inputType} in '${elem.textContent}'`);
+
+  const position = getCursorPosition(elem);
+
+  let fieldAsString = elem.textContent;
+  const overtypeLength = getOvertypeLength(event, event.data, fieldAsString, position);
+  console.log(`INPUT: '${event.data || 'N/A'}', OVERTYPE: ${overtypeLength}, POSITION: ${position}/${fieldAsString.length}`);
+  if ( overtypeLength < 0) { // Backspace (cut?)
+    if (position === fieldAsString.length) {
+      console.log(' Removing from end requires no action');
+      // Do nothing
+    }
+    else if (position < 5)  { // Replace the letter that was deleted by a backspace with a space character.
+      // NB! This presumes that overtype length is -1. Won't work for longer cuts!
+      console.log(` Replace removal with a space...\n  '${fieldAsString}`);
+      fieldAsString = `${fieldAsString.substr(0, position)} ${fieldAsString.substr(position)}`;
+      console.log(`  '${fieldAsString}'`);
+    }
+    else {
+      // NB! This currently does nothing on purpose
+      const protectedAreaSize = getProtectedAreaSize(fieldAsString);
+      if (position > 5 && position < protectedAreaSize) { // It's a datafield. We protect first subfields prefix
+        fieldAsString = `${fieldAsString.substr(0, 5)}${settings.subfieldCodePrefix}${fieldAsString.substr(protectedAreaSize)}`;
       }
-      return `<span class="value">${field.value}</span>`;
     }
-    const subfieldsToHtml = field.subfields.map(sf => `<span class="subfield"><span class="code">$$${sf.code}</span><span class="value">${sf.value}</span></span>`).join('');
-    return `<span>${subfieldsToHtml}</span>`;
   }
+  else if (overtypeLength > 0) {
```

Comment:
Could this be refactored to not have else if:s? it could be bit clearer

Reply:
Done